src/calories.rs

```diff
@@ -0,0 +1,92 @@
+#[derive(Debug, PartialEq)]
+pub struct CalorieCounter {
+    third_max: usize,
+    second_max: usize,
+    pub max: usize,
+    current_counter: usize,
+}
+
+impl CalorieCounter {
+    pub fn new(
+        third_max: usize,
+        second_max: usize,
+        max: usize,
+        current_counter: usize,
+    ) -> CalorieCounter {
+        CalorieCounter {
+            third_max,
+            second_max,
+            max,
+            current_counter,
+        }
+    }
+
+    pub fn reset() -> CalorieCounter {
+        CalorieCounter::new(0, 0, 0, 0)
+    }
+
+    pub fn with_next(&self, input: &str) -> CalorieCounter {
+        if input.is_empty() {
+            CalorieCounter::new(self.third_max, self.second_max, self.max, 0)
+        } else {
+            let input_int: usize = input.parse().unwrap();
+            let new_count = self.current_counter + input_int;
+            if new_count <= self.third_max {
+                CalorieCounter::new(self.third_max, self.second_max, self.max, new_count)
+            } else if new_count <= self.second_max {
+                CalorieCounter::new(new_count, self.second_max, self.max, new_count)
+            } else if new_count <= self.max {
+                CalorieCounter::new(self.second_max, new_count, self.max, new_count)
+            } else {
+                CalorieCounter::new(self.second_max, self.max, new_count, new_count)
+            }
```

Comment:
Isn't there a bug here?  We're updating our view of the biggest three sizes _as_ we update our view of how much the current elf is carrying.  This means that as we learn that the elf is carrying more and more stuff it might become the third, then second then finally max elf all by itself!  It's probably better to rank each elf when we read the empty terminator line (though be careful about the last elf!)

For example consider these three elves:

```
100

200

50
50
50
50
50
50
```

I think your counter will end up with

```
max: 300 // YAY
second: 250 // BOO
third: 200 // BOO2-galoo
```

The max will always be correct, but the second and third will be wrong.